src/FtpExtractor.php

```diff
@@ -54,8 +59,20 @@ public function copyFiles(string $sourcePath, string $destinationPath, FileState
         try {
             /** @var AbstractFtpAdapter $adapter */
             $adapter = $this->ftpFilesystem->getAdapter();
-            $adapter->getConnection();
-            $this->logger->info("Connected to host");
+            $logger = $this->logger;
+
+            (new RetryProxy(
+                new SimpleRetryPolicy(self::RETRY_CONNECTION, [
+                    \LogicException::class,
+                    \RuntimeException::class,
+                    \ErrorException::class,
+                ]),
+                new ExponentialBackOffPolicy(self::RETRY_BACKOFF)
+            ))->call(static function () use ($adapter, $logger): void {
+                $logger->info('Connecting to host ...');
```

Comment:
Myslel bych si, že stačí mít v retry jen to getConnection() - v logu tím pádem bude:

* Connecting to host ...
* Retrying [1x]
* Retrying [1x]

Reply:
Upraveno